### ISSUE TITLE
Event loop blocking in mock `Backend`

### ISSUE THREAD

**Issue by pavel-kokolemin** [mintlayer/mintlayer-core#631]
When mock backend receive `Command::Connect` command it will call `connect` method which will block event loop until connection succeeds, fails or timeouts, which might take a lot of time. All this time event loop won't process new events, which is bad. Ideally commands should be handled concurrently in `handle_command`.

```
    /// Runs the backend events loop.
    pub async fn run(&mut self) -> crate::Result<()> {
        loop {
            tokio::select! {
            ...
                // Handle commands.
                command = self.cmd_rx.recv() => {
                    // TODO: commands running here will block, which means (maybe) these commands in the
                    //       below function should be short-lived. The issue is that connect/disconnect
                    //       functions are part of the handling and they could take a relatively long time
                    //       we should try to figure out how to handle this problem. Maybe we need a separate
                    //       pipeline for commands that are slow, or maybe everything here should go into a new
                    //       task (which sounds excessive).
                    //       Once fixed enable self_connect_channels and self_connect_noise tests.
                    self.handle_command(command.ok_or(P2pError::ChannelClosed)?).await?;
                }
            }
        }
    }
```

**Comment by TheQuantumPhysicist** [comment]
@pavel-kokolemin Wasn't this solved in #632?

**Comment by pavel-kokolemin** [comment]
> @pavel-kokolemin Wasn't this solved in #632?

Right, it was. Thanks.
